src/gui/tray/usermodel.cpp

```diff
@@ -1013,12 +1018,31 @@ void UserModel::openCurrentAccountServer()
 
 void UserModel::setCurrentUserId(const int id)
 {
-    if (_currentUserId == id || _currentUserId < 0 || _currentUserId >= _users.size())
+    if (_currentUserId == id) {
+        // order has changed, index remained the same
+        if (id >= 0 && id < _users.size() && !_users[id]->isCurrentUser()) {
+            for (auto &user : _users) {
+                user->setCurrentUser(false);
+            }
+            _users[id]->setCurrentUser(true);
+            emit currentUserChanged();
+        }
         return;
-    
-    _users[_currentUserId]->setCurrentUser(false);
-    _users[id]->setCurrentUser(true);
+    }
     _currentUserId = id;
+
+    if (_users.isEmpty()) {
+        emit currentUserChanged();
+        return;
+    }
+
+    if (id >= 0 && id < _users.size()) {
```

Comment:
Can we not shrink this by doing:

```suggestion
    if (!_users.isEmpty() && id >= 0 && id < _users.size()) {
```